front/src/modules/ui/data/view-bar/components/MultipleFiltersDropdownFilterOnFilterChangedEffect.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { useRecoilState } from 'recoil';
+
+import { dropdownWidthState } from '@/ui/layout/dropdown/states/dropdownWidthState';
+
+export const MultipleFiltersDropdownFilterOnFilterChangedEffect = ({
+  filterDefinitionUsedInDropdownType,
+}: {
+  filterDefinitionUsedInDropdownType: string | undefined;
+}) => {
+  const [, setDropdownWidth] = useRecoilState(dropdownWidthState);

```

Comment:
so we would like this setDropdownWidth to be part of the external API of Dropdown.
Here, the developer has to know about dropdownWidthState which we want to avoid.

Could you:
- 1) add dropdownWidth getter and setter to useDropdownStates hook (this is our internal API but it makes sure to feed the state with the right "scopeId". You will find example in the code, your dropdown WidthState also needs to be refactored. The goal is to allow multiple dropdown with their own scoped state on the same page
- 2) expose this getter and setter through useDropdown (this is our external API)